### ISSUE TITLE
Add checksum to pkgfiles

### ISSUE THREAD

**Issue by PaperMountainStudio** [mitchweaver/bonsai#11]
I think it would be appropriate to add checksum (probably sha256) as a mandatory variable in pkgfiles. It should be checked against the downloaded file.
Currently, if package's server is hacked, the source file can be modified to be malicious. With checksum, this file is downloaded and then rejected.

**Comment by MitchWeaver** [comment]
I 100% agree we need this, it is one of the major TODO's on the road map.

I disagree however about it being mandatory. It will be in every package of the official ports, but in the future I want to have user's included ports, (much like any BSD `/usr/ports` system or from CRUX linux).

These user ports shouldn't be mandated to have checksums, just as they aren't in those systems either.

**Comment by PaperMountainStudio** [comment]
OK, so I think about 2 possible implementations:
1. `checksum=xxxxxxxxxxxx` in `pkgfile` (example 1)
2. file checksums in the ports/package folder which contains output of `sha256sum FILE` (example 2)

# Example 1
ports/abduco/pkgfile:
```sh
info='Tool for session {at,de}tach support'
version=0.6
source=http://github.com/martanne/abduco/archive/v$version.tar.gz
checksum=647d0381418f43a38f861d151b0efb2e3458ec651914e7d477956768b0af9bb7
```
# Example 2
ports/abduco/pkgfile:
```sh
info='Tool for session {at,de}tach support'
version=0.6
source=http://github.com/martanne/abduco/archive/v$version.tar.gz
```
ports/abduco/checksums:
```
647d0381418f43a38f861d151b0efb2e3458ec651914e7d477956768b0af9bb7  abduco#0.6.tar.gz
```

**Comment by mitchweaver** [comment]
the first would be preferred for the same reason we don't have the other variables split into different files -- less clutter

**Comment by mitchweaver** [comment]
closed by cefba0ac53f386ac0d5506382b0f5006efa12bea

preview: http://ix.io/1Qkl
